baremetal/machine_manager_test.go

```diff
@@ -0,0 +1,1626 @@
+		if err != nil {
+			t.Error(err)
+		}
+
+		err = actuator.updateMachineStatus(context.TODO(), tc.Host)
+		if err != nil {
+			t.Errorf("unexpected error %v", err)
+		}
+
+		key := client.ObjectKey{
+			Name:      tc.Machine.Name,
+			Namespace: tc.Machine.Namespace,
+		}
+
+		machine := capi.Machine{}
+		err = c.Get(context.TODO(), key, &machine)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if tc.Machine.Status.Addresses != nil {
+			for i, address := range tc.ExpectedMachine.Status.Addresses {
+				if address != machine.Status.Addresses[i] {
+					t.Errorf("expected Address %v, found %v", address, machine.Status.Addresses[i])
+				}
+			}
+		}
+	}
+}
+
+func TestApplyMachineStatus(t *testing.T) {
+	scheme := runtime.NewScheme()
+	err := capi.AddToScheme(scheme)
+
+	if err != nil {
+		log.Printf("AddToScheme failed: %v", err)
+	}
+
+	addr1 := capi.MachineAddress{
+		Type:    "InternalIP",
+		Address: "192.168.1.1",
+	}
+
+	addr2 := capi.MachineAddress{
+		Type:    "InternalIP",
+		Address: "172.0.20.2",
+	}
+
+	testCases := []struct {
+		Machine               *capi.Machine
+		Addresses             []capi.MachineAddress
+		ExpectedNodeAddresses []capi.MachineAddress
+	}{
+		{
+			// Machine status updated
+			Machine: &capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine1",
+					Namespace: "myns",
+				},
+				Status: capi.MachineStatus{
+					Addresses: []capi.MachineAddress{},
+				},
+			},
+			Addresses:             []capi.MachineAddress{addr1, addr2},
+			ExpectedNodeAddresses: []capi.MachineAddress{addr1, addr2},
+		},
+		{
+			// Machine status unchanged
+			Machine: &capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine1",
+					Namespace: "myns",
+				},
+				Status: capi.MachineStatus{
+					Addresses: []capi.MachineAddress{addr1, addr2},
+				},
+			},
+			Addresses:             []capi.MachineAddress{addr1, addr2},
+			ExpectedNodeAddresses: []capi.MachineAddress{addr1, addr2},
+		},
+	}
+
+	for _, tc := range testCases {
+		c := fakeclient.NewFakeClientWithScheme(scheme)
+		if tc.Machine != nil {
+			err := c.Create(context.TODO(), tc.Machine)
+			if err != nil {
+				t.Errorf("Create failed, %v",
+					err)
+			}
+		}
+		actuator, err := newMachineManager(c, nil, nil, tc.Machine, nil)
+		if err != nil {
+			t.Error(err)
+		}
+
+		err = actuator.applyMachineStatus(context.TODO(), tc.Machine, tc.Addresses)
+		if err != nil {
+			t.Errorf("unexpected error %v", err)
+		}
+
+		key := client.ObjectKey{
+			Name:      tc.Machine.Name,
+			Namespace: tc.Machine.Namespace,
+		}
+		machine := capi.Machine{}
+		err = c.Get(context.TODO(), key, &machine)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if tc.Machine.Status.Addresses != nil {
+			for i, address := range tc.ExpectedNodeAddresses {
+				if address != machine.Status.Addresses[i] {
+					t.Errorf("expected Address %v, found %v", address, machine.Status.Addresses[i])
+				}
+			}
+		}
+	}
+}
```

Comment:
The applyMachineStatus function is unused, so it and this test could be removed

Reply:
done